src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,19 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          let focusableDescendants = 0;
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) {
+              focusableDescendants += 1;
+            }
+            if (focusableDescendants >= 2 && child === document.activeElement) {
```

Comment:
maybe adding a small comment around why `2` is the condition would be helpful for folks in future.

suggestion but feel free to alter
```
// FormField descendant could be focusable. If descendant is
// active element in DOM, don't show focus on FormField.
// e.g. DateInput allows Calendar button to receive focus
```

Reply:
Added a comment in the latest push